mycpp/gc_str.cc

```diff
@@ -482,3 +488,96 @@ List<Str*>* Str::split(Str* sep) {
 
   return result;
 }
+
+// TODO: We should reduce the number of allocations and copies done here.
+static inline Str* _StrFromFormat(const char* fmt, va_list args) {
+  int n = strlen(fmt);
+  auto beg = std::cregex_iterator(fmt, fmt + n, gStrFmtRegex);
+  auto end = std::cregex_iterator();
+  mylib::BufWriter* w = Alloc<mylib::BufWriter>();
+  for (std::cregex_iterator match = beg; match != end; ++match) {
+    auto& lit = (*match)[1];
```

Comment:
for readability I think NOT using auto would be better, since I don't know this API

IMO it's nice to have the type shown exactly once, not zero or 2 times